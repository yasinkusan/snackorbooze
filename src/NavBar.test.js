import React from 'react';
import NavBar from './NavBar';
import { MemoryRouter } from 'react-router-dom';

import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

function sleep(time) {
	return new Promise((resolve) => setTimeout(resolve, time));
}

it('renders without crashing', () => {
	render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<NavBar />
		</MemoryRouter>
	);
});

it('matches loading snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<NavBar />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
it('matches rendered snapshot', async () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<NavBar />
		</MemoryRouter>
	);
	await act(() => sleep(1100));
	expect(asFragment()).toMatchSnapshot();
});
