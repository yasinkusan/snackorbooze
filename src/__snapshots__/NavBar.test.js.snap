// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches loading snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/new"
          >
            Add Menu Item
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches rendered snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/new"
          >
            Add Menu Item
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
